pkg/apis/serving/v1alpha2/framework_xgboost.go

```diff
@@ -36,6 +37,11 @@ func (x *XGBoostSpec) GetResourceRequirements() *v1.ResourceRequirements {
 }
 
 func (x *XGBoostSpec) GetContainer(modelName string, hasLogging bool, config *InferenceServicesConfig) *v1.Container {
+	nthread := x.NThread
```

Comment:
Sorry to thrash you -- we're almost there. I had one more thought on where to plug in particular behavior.

This is fine and works, but there's a slightly better spot. I'd probably use our spec defaulter to set this value: https://github.com/kubeflow/kfserving/blob/master/pkg/apis/serving/v1alpha2/framework_xgboost.go#L51. It's a little tricky, because it means that if a user applies a spec with 4 cores and then upgrades it to 8 cores, it will only default once.

It ends up making it more explicit and controlled (i.e., the user can see their nthreads default), but it could be a little awkward if cores gets increased but threads don't. I think it would be reasonably obvious to figure out though, as we'd be exposing this concept to users by nature of defaulting.

Happy with either approach, leaning doing the defaulting mechanism instead. WDYT?

Reply:
Yep I agree to put it into `ApplyDefaults` too

Reply:
Updated in 5f82f38